src/other/codemirror/utils/exaxyz_linter.ts

```diff
@@ -0,0 +1,30 @@
+import { Diagnostic } from "@codemirror/lint";
+import { EditorView } from "@codemirror/view";
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { ConsistencyCheck } from "@exabyte-io/code.js/dist/types";
+import _ from "underscore";
+
+const linterGenerator = (checks: ConsistencyCheck[]) => {
+    return (view: EditorView): Diagnostic[] => {
+        const { doc } = view.state;
+        if (!checks && !doc) return [];
+
+        const warnings = checks.map((check) => {
+            const keyFragments = check.key.split(".");
+            const atomIdStr = _.last(keyFragments);
```

Comment:
AtomId is a Material domain - it should not be in Cove, which is generic.

linterGenerator, in principle, is a generic enough to be here

Reply:
Yes. Basically, it's just the difference in naming. The logic itself is generic already. renaming: AtomId -> key